ui/components/app/edit-gas-popover/edit-gas-popover.component.js

```diff
@@ -55,26 +93,43 @@ export default function EditGasPopover({
     if (!transaction || !mode) {
       closePopover();
     }
+
+    const cancelSpeedUpGas =
+      gasEstimateType === GAS_ESTIMATE_TYPES.FEE_MARKET
+        ? {
+            gasLimit: decGWEIToHexWEI(gasLimit),
+            maxFeePerGas: decGWEIToHexWEI(maxFeePerGas),
+            maxPriorityFeePerGas: decGWEIToHexWEI(maxPriorityFeePerGas),
+          }
+        : {
+            gasLimit: decGWEIToHexWEI(gasLimit),
+            gasPrice: decGWEIToHexWEI(gasPrice),
+          };
+
     switch (mode) {
-      case EDIT_GAS_MODE.CANCEL:
-        dispatch(
-          createCancelTransaction(transaction.id, {
-            /** new gas settings */
-          }),
-        );
+      case EDIT_GAS_MODES.CANCEL:
+        dispatch(createCancelTransaction(transaction.id, cancelSpeedUpGas));
         break;
-      case EDIT_GAS_MODE.SPEED_UP:
-        dispatch(
-          createSpeedUpTransaction(transaction.id, {
-            /** new gas settings */
-          }),
-        );
+      case EDIT_GAS_MODES.SPEED_UP:
+        dispatch(createSpeedUpTransaction(transaction.id, cancelSpeedUpGas));
         break;
-      case EDIT_GAS_MODE.MODIFY_IN_PLACE:
+      case EDIT_GAS_MODES.MODIFY_IN_PLACE:
         dispatch(
           updateTransaction({
             ...transaction,
-            txParams: { ...transaction.txParams /** ...newGasSettings */ },
+            txParams: {
+              ...transaction.txParams,
+              ...(process.env.SHOW_EIP_1559_UI
```

Comment:
this is essentially the same as the cancelSpeedUpGas, so you can use it here too, just add the gas key manually. it won't hurt anything to add both gasLimit and gas (only one will be used)

Reply:
Good to know!  Updated!